src/test/rbf_tests.cpp

```diff
@@ -0,0 +1,211 @@
+    const auto entry4 = pool.GetIter(tx4->GetHash()).value();
+    const auto entry5 = pool.GetIter(tx5->GetHash()).value();
+    const auto entry6 = pool.GetIter(tx6->GetHash()).value();
+    const auto entry7 = pool.GetIter(tx7->GetHash()).value();
+    const auto entry8 = pool.GetIter(tx8->GetHash()).value();
+
+    BOOST_CHECK_EQUAL(entry1->GetFee(), normal_fee);
+    BOOST_CHECK_EQUAL(entry2->GetFee(), normal_fee);
+    BOOST_CHECK_EQUAL(entry3->GetFee(), low_fee);
+    BOOST_CHECK_EQUAL(entry4->GetFee(), high_fee);
+    BOOST_CHECK_EQUAL(entry5->GetFee(), low_fee);
+    BOOST_CHECK_EQUAL(entry6->GetFee(), low_fee);
+    BOOST_CHECK_EQUAL(entry7->GetFee(), high_fee);
+    BOOST_CHECK_EQUAL(entry8->GetFee(), high_fee);
+
+    CTxMemPool::setEntries set_12_normal{entry1, entry2};
+    CTxMemPool::setEntries set_34_cpfp{entry3, entry4};
+    CTxMemPool::setEntries set_56_low{entry5, entry6};
+    CTxMemPool::setEntries all_entries{entry1, entry2, entry3, entry4, entry5, entry6, entry7, entry8};
+    CTxMemPool::setEntries empty_set;
+
+    const auto unused_txid{GetRandHash()};
+
+    // Tests for PaysMoreThanConflicts
+    // These tests use feerate, not absolute fee.
+    BOOST_CHECK(PaysMoreThanConflicts(/*iters_conflicting=*/set_12_normal,
+                                      /*replacement_feerate=*/CFeeRate(entry1->GetModifiedFee() + 1, entry1->GetTxSize() + 2),
+                                      /*txid=*/unused_txid).has_value());
+    // Replacement must be strictly greater than the originals.
+    BOOST_CHECK(PaysMoreThanConflicts(set_12_normal, CFeeRate(entry1->GetModifiedFee(), entry1->GetTxSize()), unused_txid).has_value());
+    BOOST_CHECK(PaysMoreThanConflicts(set_12_normal, CFeeRate(entry1->GetModifiedFee() + 1, entry1->GetTxSize()), unused_txid) == std::nullopt);
+    // These tests use modified fees (including prioritisation), not base fees.
+    BOOST_CHECK(PaysMoreThanConflicts({entry5}, CFeeRate(entry5->GetModifiedFee() + 1, entry5->GetTxSize()), unused_txid) == std::nullopt);
+    BOOST_CHECK(PaysMoreThanConflicts({entry6}, CFeeRate(entry6->GetFee() + 1, entry6->GetTxSize()), unused_txid).has_value());
+    BOOST_CHECK(PaysMoreThanConflicts({entry6}, CFeeRate(entry6->GetModifiedFee() + 1, entry6->GetTxSize()), unused_txid) == std::nullopt);
+    // These tests only check individual feerate. Ancestor feerate does not matter.
+    BOOST_CHECK(PaysMoreThanConflicts(set_34_cpfp, CFeeRate(entry4->GetModifiedFee(), entry4->GetTxSize()), unused_txid).has_value());
+
+    // Tests for EntriesAndTxidsDisjoint
+    BOOST_CHECK(EntriesAndTxidsDisjoint(empty_set, {tx1->GetHash()}, unused_txid) == std::nullopt);
+    BOOST_CHECK(EntriesAndTxidsDisjoint(set_12_normal, {tx3->GetHash()}, unused_txid) == std::nullopt);
+    // EntriesAndTxidsDisjoint uses txids, not wtxids.
+    BOOST_CHECK(EntriesAndTxidsDisjoint({entry2}, {tx2->GetWitnessHash()}, unused_txid) == std::nullopt);
```

Comment:
```suggestion
    BOOST_CHECK(EntriesAndTxidsDisjoint({entry2}, {tx2->GetWitnessHash()}, unused_txid) == std::nullopt);
    BOOST_CHECK(EntriesAndTxidsDisjoint({entry2}, {tx2->GetHash()}, unused_txid).has_value());
```
rules out the case where tx2 has no witness data(same wtxid) but some other bug.,.

Reply:
Done